geckomat/utilities/ecFSEOF/run_ecFSEOF.m

```diff
@@ -0,0 +1,69 @@
+function results = run_ecFSEOF(model,rxnTarget,cSource,alphaLims,Nsteps,file1,file2)
+% 
+% run_ecFSEOF
+%
+% 	Function that runs Flux-scanning with Enforced Objective Function
+% 	for a specified production target.
+%   
+%		model     (struct) ecModel with total protein pool constraint
+%		rxnTarget (string) Rxn ID for the production target reaction, 
+% 				  a exchange reaction is recommended.
+%		cSource	  (string) Rxn name for the main carbon source uptake 
+%			      reaction
+% 		alphaLims (vector) Minimum and maximum biomass yield [gDw/mmol Csource] 
+%				  for enforced objective limits
+%       Nsteps    (integer) Number of steps for suboptimal objective 
+%                 in FSEOF
+%		file1 	  (string) opt, file name for an output .txt tab-separated 
+% 				  file for the results at the genes level
+%		file2 	  (string) opt, file name for an output .txt tab-separated 
+% 				  file for the results at the reactions level		
+%
+% Usage: run_ecFSEOF(ecModel,rxnTarget,cSource,alphaLims,Nsteps)
+%
+% Last modified.  Ivan Domenzain 2019-11-13
+%
+
+if nargin < 7
+	file2 = [];
+	if nargin < 6
+		file1 = [];
+	end
+end
+%Define alpha vector for suboptimal enforced objective values
+alphaV  = alphaLims(1):((alphaLims(2)-alphaLims(1))/(Nsteps-1)):alphaLims(2);
+%Standardize grRules and rxnGeneMat in model
+[grRules,rxnGeneMat] = standardizeGrRules(model,true);
+model.grRules        = grRules;
+model.rxnGeneMat     = rxnGeneMat;
+%run FSEOF analysis
+results = ecFlux_scanning(model,rxnTarget,cSource,alphaV,1E-4,true);
+%Create gene table:
+results.geneTable      = cell(length(results.genes),3);
+results.geneTable(:,1) = results.genes;
+results.geneTable(:,2) = results.geneNames;
+results.geneTable(:,3) = num2cell(results.k_genes);
+%Create rxns table (exclude enzyme usage reactions):
+enzIndxs               = find(~startsWith(results.rxns(:,1),'draw_prot_'));
+results.k_rxns         = results.k_rxns(enzIndxs);
+results.rxnsTable      = cell(length(results.k_rxns),5);
+results.rxnsTable(:,1) = results.rxns(enzIndxs,1);
+results.rxnsTable(:,2) = results.rxns(enzIndxs,2);
+results.rxnsTable(:,3) = num2cell(results.k_rxns);
+results.rxnsTable(:,4) = results.rxns(enzIndxs,3);
+results.rxnsTable(:,5) = results.rxns(enzIndxs,4);
```

Comment:
If you are creating the summaries `rxnsTable` and `geneTable`, then I would suggest removing from the results structure `k_rxns`, `rxns`, `genes`, `geneNames` and `k_genes`, as they are all redundant information.

Also, consider filtering `v_matrix` and `k_matrix` as well, to avoid having reaction fields of different sizes:

![image](https://user-images.githubusercontent.com/9384349/104010799-60ca6100-51ad-11eb-8c9f-f81538c7bcf3.png)

Reply:
I agree. Both suggestions have been addressed in: fc0df92 and  291640f